pvm.py

```diff
@@ -9,84 +10,88 @@
-	handler.setLevel(logging.INFO)
-	formatter = logging.Formatter("%(asctime)s.%(msecs)03d;%(levelname)s;%(message)s",
-							  "%Y-%m-%d %H:%M:%S")
-	fileHandler.setFormatter(formatter)
-	logger.addHandler(fileHandler)
-	handler.setFormatter(formatter)
-	logger.addHandler(handler)
-
-_init_logger()
-_logger = logging.getLogger("PVM")
-_logger.info("Logging system initiated in %s", LOG_PATH)
-
-media = ""
+LOG_PATH = HOME + "PVM/log/" # Log path
+PREFIX_VIDEOS_PATH = HOME + "Videos/" # Place your videos in this folder for autostart
+VIDEO_PATH = None # Video path
+OMX = None	# OMXPlayer object
+IS_FILE_SET = False # File set flag
+CAN_PAUSE = False # Can pause flag
+CAN_START = False # Can start flag
 
 '''
 UDP command example:
 /PVM HH MM SS <command> <value>
 '''
-
 def parse_commands(*args):
-	global media
+	global OMX
+	global PREFIX_VIDEOS_PATH
 	global VIDEO_PATH
 	global IS_FILE_SET
-	
-	# Parse time and add 3 seconds
-	hh, mm, ss = args[1], args[2], args[3]
-	time_str = str(hh) + ":" + str(mm) + ":" + str(ss)
-	next_time = datetime.strptime(time_str, "%H:%M:%S") + timedelta(seconds=3)
+	global CAN_START
+	global CAN_PAUSE
+	global PORT
 
 	# Get command
 	command = args[4]
+	
+	# Only start and pause command need to delay 3s
+	if command == "start" or command == "pause":
+		# Parse time and add 3 seconds
+		hh, mm, ss = args[1], args[2], args[3]
+		time_str = str(hh) + ":" + str(mm) + ":" + str(ss)
+		next_time = datetime.strptime(time_str, "%H:%M:%S") + timedelta(seconds=3)
 
-	# If scheduled time is behind current time, return.
-	if next_time.time() <= datetime.now().time():
-		_logger.info("Command: %s failed because scheduled time(%s) is behind current time.", command, time_str)
-		return
+		# If scheduled time is behind current time, return.
+		# In very rare cases, the control computer will have a three-second time difference with the Raspberry Pi
+		if next_time.time() <= datetime.now().time():
+			_logger.info("Command: %s failed because scheduled time(%s) is behind current time.", command, time_str)
+			return
 
-	# Log command and execute time.
-	_logger.info("Command: %s, execute time: %s.", command, next_time.time())
-	
-	# Wait until current time is equal to next_time
-	while True:
-		now = datetime.now()
-		now = now.replace(microsecond=0)
-		if now.time() >= next_time.time():
-			break
-		sleep(0.005)
-	
-    # Get value
-	if len(args) == 6:
+		# Log command and execute time.
+		_logger.info("Command: %s, execute time: %s.", command, next_time.time())
+		
+		# Wait until current time is equal to next_time
+		while True:
+			now = datetime.now()
+			now = now.replace(microsecond=0)
+			if now.time() >= next_time.time():
+				break
+			sleep(0.005)
+
+	# Get value
+	if len(args) >= 6:
 		value = args[5]
 
+	# If PORT == 8001, we export our video in the main display 2,
+	# else we export our video on the extra display 7
+	params = ['--loop', '--display']
+	if PORT == 8001:
+		params.append('2')
```

Comment:
this needs to be explained in the readme

Reply:
Done